src/ophyd_async/core/signal.py

```diff
@@ -196,18 +196,18 @@ async def unstage(self) -> None:
 USE_DEFAULT_TIMEOUT = "USE_DEFAULT_TIMEOUT"
 
 
-class SignalW(Signal[T], Movable):
+class SignalW(Signal[S, S], Movable):
     """Signal that can be set"""
 
-    def set(self, value: T, wait=True, timeout=USE_DEFAULT_TIMEOUT) -> AsyncStatus:
+    def set(self, value: S, wait=True, timeout=USE_DEFAULT_TIMEOUT) -> AsyncStatus:
         """Set the value and return a status saying when it's done"""
         if timeout is USE_DEFAULT_TIMEOUT:
             timeout = self._timeout
         coro = self._backend.put(value, wait=wait, timeout=timeout)
         return AsyncStatus(coro)
 
 
-class SignalRW(SignalR[T], SignalW[T], Locatable):
+class SignalRW(SignalR[R], SignalW[S], Locatable):
```

Comment:
It's a shame to have this for all signals when `R` and `S` will be the same >90% of the time. I wonder if it would make things simpler to introduce a new type instead of coopting `SignalRW`? @coretl may have thoughts.